src/Compilers/CSharp/Portable/Symbols/TypeWithState.cs

```diff
@@ -82,18 +84,24 @@ private TypeWithState(TypeSymbol type, NullableFlowState state)
 
         public TypeWithState WithSuppression(bool suppress) => suppress ? new TypeWithState(Type, NullableFlowState.NotNull) : this;
 
-        public TypeWithAnnotations ToTypeWithAnnotations()
+        public TypeWithAnnotations ToTypeWithAnnotations(CSharpCompilation compilation, bool asAnnotatedType = false)
         {
-            NullableAnnotation annotation = this.State.IsNotNull() || Type?.CanContainNull() == false || Type?.IsTypeParameterDisallowingAnnotation() == true
-                ? NullableAnnotation.NotAnnotated : NullableAnnotation.Annotated;
+            if (Type is null)
+            {
+                return TypeWithAnnotations.Create(Type, NullableAnnotation.Annotated);
```

Comment:
```suggestion
                return TypeWithAnnotations.Create(null, NullableAnnotation.Annotated);
```

Why do we unconditionally use `NullableAnnotation.Annotated` here for `null` types?